drape_frontend/stylist.cpp

```diff
@@ -215,20 +215,16 @@ Stylist::Stylist(FeatureType & f, uint8_t zoomLevel, int8_t deviceLang)
     if (m_captionDescriptor.IsHouseNumberExists())
     {
       bool isGood = true;
-      if (zoomLevel < scales::GetUpperStyleScale())
+      if (zoomLevel < scales::GetPointHousenumbersScale())
       {
         if (geomType == feature::GeomType::Area)
         {
           // Don't display housenumbers when an area (e.g. a building) is too small.
           m2::RectD const r = f.GetLimitRect(zoomLevel);
-          isGood = std::min(r.SizeX(), r.SizeY()) > scales::GetEpsilonForHousenumbers(zoomLevel);
+          isGood = std::max(r.SizeX(), r.SizeY()) > scales::GetEpsilonForHousenumbers(zoomLevel);
```

Comment:
Better to keep it as is.
Otherwise it'll make long narrow buildings display the number while square bigger houses nearby won't have it.

Reply:
This is what I mean. There are visually bigger houses around but only a narrow one has a number.

![image](https://github.com/organicmaps/organicmaps/assets/18434508/2395afc8-bf14-472d-8d2a-6d511f264cf3)

![image](https://github.com/organicmaps/organicmaps/assets/18434508/84e6f285-d936-4f4e-a6f1-0e0c3c5d4cf9)